src/bitcoin/checkpoint.rs

```diff
@@ -2200,6 +2318,49 @@ impl CheckpointQueue {
 
         Ok(Some(self.index - num_unconf - signing_offset))
     }
+
+    pub fn unconfirmed(&self) -> Result<Vec<Ref<'_, Checkpoint>>> {
+        let first_unconf_index = self.first_unconfirmed_index()?;
+        if let Some(index) = first_unconf_index {
+            let mut out = vec![];
+            for i in index..=self.index {
+                let cp = self.get(i)?;
+                if !matches!(cp.status, CheckpointStatus::Complete) {
+                    break;
+                }
+                out.push(cp);
+            }
+            Ok(out)
+        } else {
+            Ok(vec![])
+        }
+    }
+
+    pub fn unconfirmed_fees_paid(&self) -> Result<u64> {
+        self.unconfirmed()?
+            .iter()
+            .map(|cp| cp.checkpoint_tx_miner_fees())
+            .try_fold(0, |fees, result: Result<_>| {
+                let fee = result?;
+                Ok::<_, Error>(fees + fee)
+            })
+    }
+
+    pub fn unconfirmed_vbytes(&self) -> Result<u64> {
+        self.unconfirmed()?
+            .iter()
+            .map(|cp| Ok(cp.checkpoint_tx()?.vsize() as u64))
```

Comment:
this isn't counting the witness, causing `fee_adjustment()` to return 0 even with a long chain of unconfirmed checkpoints. should use `est_size()`.